standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/dataconnector/jdbc/AbstractJDBCConnectorProvider.java

```diff
@@ -117,7 +117,7 @@ public AbstractJDBCConnectorProvider(String dbName, DataConnector dataConn, Stri
 
   protected Connection getConnection() {
     try {
-      if (!isOpen)
+      if (!isOpen())
         open();
```

Comment:
Please close the invalid connection first

Reply:
Done. I make sure close the invalid connection in method `isClosed()`.

Reply:
cloud we move it in the `open()` method?
`open() {
  close()
   // open new connection
}`

Reply:
Done